wgpu-hal/src/metal/device.rs

```diff
@@ -75,6 +75,7 @@ impl super::Device {
 
         let options = mtl::CompileOptions::new();
         options.set_language_version(self.shared.private_caps.msl_version);
+        options.set_preserve_invariance(true);
```

Comment:
this is constrained on the OS - https://developer.apple.com/documentation/metal/mtlcompileoptions/3564462-preserveinvariance
So we can't call it without checking, and this check should be somewhere in `PrivateCapabilities`

Reply:
Aww schucks, I should of checked that first